### PR TITLE
WIP: Multiplayer support 🚧 🏗

### PR DESCRIPTION
**Pull request by chrisheninger**
Currently running into an issue, socket debug looks like it may be this?

```
socket.io-parser decoded 4/matchimals,"Invalid namespace" as {"type":4,"nsp":"/matchimals","data":"Invalid namespace"} +2ms
```

TODO:
- [ ] Connect to server, make sure it syncs state between multiple browser windows
- [ ] Add UI/logic/routes to join a room
- [ ] Add logic that automatically puts player in a seat when they connect to a room– add UI to indicate "waiting on players" as well as spectator mode if room is full on players
- [ ] Add UI/logic for when a player disconnects from the room– "wait for another player" or "leave game"

### CODE REVIEW

**Comment by Stefan-Hanke** [comment]
Could you please add repro steps? I've tried the following:
 - `npm run server` - listening on port 3333
 - did a clone of branch multiplayer, installed deps, copied it, and modified one to have `playerID=1` in its `src/index.js`.
 - run both, one on port 3000 and the other on port 3001

It looks like this opens up two different game instances. `index.js` on both sides contains the reference to `localhost:3333`.

**Comment by chrisheninger** [comment]
@Stefan-Hanke Yeah– those are the exact steps that I was using.

I noticed that the game wasn't syncing like I'd expect it to. Upon trying to learn how to diagnose a socket connection, I found that `localStorage.debug = '*';` in your console will make it automatically log all the socket-related info. From that output I found the "Invalid Namespace" error– it could be something completely unrelated but that's all I saw that looked strange.

File: server/index.js [hunk 1]
```diff
@@ -0,0 +1,6 @@
+const Server = require('boardgame.io/server');
+const Matchimals = require('../src/Game');
```
Comment (nicolodavis) [inline server/index.js]: I think the problem is here. You may have to do:

```
const Matchimals = require('../src/Game').default;
```

**Comment by chrisheninger** [comment]
Thanks both of you– I should have a chance in the coming weeks to give your recommendations a try. Appreciate your time.

**Comment by chrisheninger** [comment]
Closing for now, will re-open and implement once the native app is launched.

**Pull request closed without merging**